storage/range_test.go

```diff
@@ -2717,3 +2717,47 @@ func BenchmarkWriteCmdWithEvents(b *testing.B) {
 func BenchmarkWriteCmdWithEventsAndConsumer(b *testing.B) {
 	benchmarkEvents(b, true, true)
 }
+
+type mockRangeManager struct {
+	*Store
+	mockProposeRaftCommand func(cmdIDKey, proto.InternalRaftCommand) <-chan error
+}
+
+// ProposeRaftCommand implement the ProposeRaftCommand method of rangeManager.
+func (mrm *mockRangeManager) ProposeRaftCommand(idKey cmdIDKey, cmd proto.InternalRaftCommand) <-chan error {
+	if mrm.mockProposeRaftCommand == nil {
+		return mrm.Store.ProposeRaftCommand(idKey, cmd)
+	}
+	return mrm.mockProposeRaftCommand(idKey, cmd)
+}
+
+// TestRequestLeaderEncounterGroupDeleteError verifies ErrGroupDeleted returned by AddCmd
+// should be converted to RangeNotFoundError.
```

Comment:
> 
// TestRaftGroupDeletedError verifies that a proposal which fails with multiraft.ErrGroupDeleted 
// is converted to a RangeNotFoundError in the Store.